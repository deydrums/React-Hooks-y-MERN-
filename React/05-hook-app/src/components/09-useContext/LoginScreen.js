import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {
    /**1.Obtener la referencia al userContext
     * 2.setUser
     * {
     *  id:123,
     *  name: David
     * }
     * **/

     const {setUser} = useContext(UserContext);
     
    return (
        <div>
            <h1>LoginScreen</h1>
            <hr></hr>
            <button
                className="btn btn-success"
                onClick={()=>setUser({
                    id: 123,
                    name: 'David',
                    email: 'dagarcia100@gmail.com'
                })}
            >
                Login
            </button>
        </div>
    )
}
