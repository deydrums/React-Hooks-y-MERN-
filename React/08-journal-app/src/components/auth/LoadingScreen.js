import React from 'react'

export const LoadingScreen = () => {
    return (
        <div>
            <div className="auth__main">
                <div className="auth__loading">
                <div class="page-loader">
                    <div></div>
                    <div></div>
                    <div></div>
                </div>
                </div>
            </div>
        </div>
    )
}
