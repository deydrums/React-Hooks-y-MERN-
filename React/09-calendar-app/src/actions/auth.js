import { fetchWithoutToken } from "../helpers/fetch"
import { types } from "../types/types";
import Swal from 'sweetalert2';



export const startLogin = (email, password) => {
    return async(dispatch) => {
        const resp = await fetchWithoutToken('auth',{email,password},'POST');
        const body = await resp.json();
        console.log(body);
        if(resp.ok) {
            localStorage.setItem('token',body.token);
            localStorage.setItem('token-init-date', new Date().getTime());
            dispatch(login({
                uid: body.uid,
                name: body.name
            }));
        }else{
            Swal.fire('Error',body.message?body.message:'Ha ocurrido un error','error');
        }
    }
}

export const startRegister = (name, email, password) => {
    return async(dispatch) => {
        const resp = await fetchWithoutToken('auth/new',{name,email,password},'POST');
        const body = await resp.json();
        console.log(body);
        if(resp.ok) {
            localStorage.setItem('token',body.token);
            localStorage.setItem('token-init-date', new Date().getTime());
            dispatch(login({
                uid: body.uid,
                name: body.name
            }));
        }else{
            Swal.fire('Error',body.message?body.message:'Ha ocurrido un error','error');
        }
    }
}

const login = (user) => ({
    type: types.authLogin,
    payload: user
});