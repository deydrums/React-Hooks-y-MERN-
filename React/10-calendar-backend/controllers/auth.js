const {response} = require('express');
const User = require('../models/User');
var bcrypt = require('bcryptjs');
const {generateJWT} = require('../helpers/jwt');

const createUser = async(req,res = response)=>{
    const {email, password} = req.body;
    try {
        let user = await User.findOne({email: email});
        if(user){
            return res.status(400).json({
                ok: false,
                message: 'El correo ya fue registrado anteriormente'
            });
        };
        user = new User(req.body);
        //Encriptar contraseÃ±a
        const salt = bcrypt.genSaltSync();
        user.password = bcrypt.hashSync(password,salt);

        await user.save();
        //Generar nuestro JWT
        const token = await generateJWT(user.id, user.name);

    
        res.status(201).json({
            ok: true,
            message: 'Registro de usuario exitoso',
            uid: user.id,
            name: user.name,
            token
        });

    } catch (error) {
        console.log(error);
        res.status(500).json({
            ok: false,
            message: 'Ha ocurrido un error, intenta de nuevo'
        })
    }
}

const loginUser = async(req,res = response)=>{
    const {email, password} = req.body

    try {
        const user = await User.findOne({email: email});
        if(!user){
            return res.status(404).json({
                ok: false,
                message: 'Usuario no encontrado'
            });
        };

        //Confirmar los passwords
        const validPassword = bcrypt.compareSync(password, user.password);
        if(!validPassword){
            return res.status(400).json({
                ok: false,
                message: 'Password incorrecto'
            });
        };

        //Generar nuestro JWT
        const token = await generateJWT(user.id, user.name);

        res.status(200).json({
            ok: true,
            message: 'Login correcto',
            uid: user.id,
            name: user.name,
            token
        });


    } catch (error) {
        console.log(error);
        res.status(500).json({
            ok: false,
            message: 'Ha ocurrido un error, intenta de nuevo'
        });
    }
}

const renewToken = async(req,res = response)=>{
    const uid = req.uid;
    const name = req.name;

    //Generar nuestro JWT
    const token = await generateJWT(uid, name);

    res.json({
        ok: true,
        message: 'Nuevo token generado',
        uid,
        name,
        token
    })
}

module.exports = {
    createUser,
    loginUser,
    renewToken
};