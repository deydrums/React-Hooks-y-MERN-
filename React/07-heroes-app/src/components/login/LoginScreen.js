import React, { useContext, useReducer } from 'react'
import { AuthContext } from '../../auth/AuthContext';
import { authReducer } from '../../auth/authReducer';
import { types } from '../../types/types';
import logo from './logo.svg';

export const LoginScreen = ({history}) => {

    const {dispatch} = useContext(AuthContext);

    const handleLogin = () => {
        const action = {
            type: types.login,
            payload: {
                name: 'Invitado',
                email: 'invitado@invitado.com'
            }
        };
        //Dispatch
        dispatch(action);
        history.replace('/');

    };

    return (
        <>
        <div className="header" id="inicio">
            <div className="login">
                <h1>HeroesApp</h1>
                <img src={logo} className="App-logo" alt="logo" />
                <button
                    className="login-button"
                    onClick={handleLogin}
                >
                    Entrar
                </button>

            </div>
            <footer className="footer-login">Powered by <a href="http://deydrums.com">David Garcia</a></footer>

        </div>


        </>
    )
}
