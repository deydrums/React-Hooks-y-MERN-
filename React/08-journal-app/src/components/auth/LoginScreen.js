import React from 'react';
import {Link} from 'react-router-dom';
import { useForm } from '../../hooks/useForm';
import { useDispatch } from 'react-redux';
import { login } from '../../actions/auth';

export const LoginScreen = () => {

    const dispatch = useDispatch();

    const [formValues, handleInputChange, reset] = useForm({
        email: 'admin@admin.com',
        password: '123456'
    });

    const {email, password} = formValues;

    const handleLogin = (e) => {
        e.preventDefault();
        dispatch(login(12345,'David'));
    }

    return (
        <>
            <h3 className="auth__title">Iniciar sesión</h3>
            <form onSubmit={handleLogin}>
                <input
                    type="text"
                    placeholder="Email..."
                    name = "email"
                    className="auth__input"
                    autoComplete="off"
                    value = {email}
                    onChange={handleInputChange}
                />
                <input
                    type="password"
                    placeholder="Password..."
                    name = "password"
                    className="auth__input"
                    value = {password}
                    onChange={handleInputChange}
                />
                <button
                    type="submit"
                    className = "btn btn-primary btn-block"
                >
                    Entrar
                </button>
                
                <div className="auth__social-networks">
                    <p>Iniciar con redes sociales</p>
                    <div 
                        className="google-btn"
                    >
                        <div className="google-icon-wrapper">
                            <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
                        </div>
                        <p className="btn-text">
                            <b>Entrar con Google</b>
                        </p>
                    </div>
                </div>
                <Link 
                    to="/auth/register"
                    className="link"
                >
                    Registrarse aquí
                </Link>
            </form>
        </>
    )
}
