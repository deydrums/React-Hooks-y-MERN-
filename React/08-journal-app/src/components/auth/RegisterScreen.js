import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {Link} from 'react-router-dom';
import validator from 'validator';
import { removeError, setError } from '../../actions/ui';
import { useForm } from '../../hooks/useForm';

export const RegisterScreen = () => {

    const dispatch = useDispatch();
    const {msgError} = useSelector(state =>state.ui);

    const [formValues, handleInputChange, reset] = useForm({
        name: 'admin',
        email: 'admin@admin.com',
        password: '123456',
        password2: '123456'
    });

    const {name, email, password, password2} = formValues;

    const handleRegister = (e) => {
        e.preventDefault();
        if (isFormValid()){
            console.log('Formulario Correcto')
        }
    };

    const isFormValid = () =>{
        if(name.trim().length ===0){
            dispatch(setError('Nombre es requerido'));
            return false;
        }else if(!validator.isEmail(email)){
            dispatch(setError('Email invalido'));
            return false;
        }else if (password !== password2 || password.length < 5){
            dispatch(setError('Contraseñas no coinciden (deben ser mayor a 4 caracteres)'));
            return false;
        }
        dispatch(removeError());
        return true;
    };

    return (
        <>
            <h3 className="auth__title">Registrarse</h3>
            {            
                msgError &&
                <div className="auth__alert-error"> 
                    {msgError}
                </div>
            }            
            <form onSubmit = {handleRegister}>
                <input
                    type="text"
                    placeholder="Nombre..."
                    name = "name"
                    className="auth__input"
                    autoComplete="off"
                    onChange = {handleInputChange}
                    value = {name}
                />
                <input
                    type="text"
                    placeholder="Email..."
                    name = "email"
                    className="auth__input"
                    autoComplete="off"
                    onChange = {handleInputChange}
                    value = {email}
                />
                <input
                    type="password"
                    placeholder="Contraseña..."
                    name = "password"
                    className="auth__input"
                    onChange = {handleInputChange}
                    value = {password}
                />
                <input
                    type="password"
                    placeholder="Confirmar contraseña..."
                    name = "password2"
                    className="auth__input"
                    onChange = {handleInputChange}
                    value = {password2}
                />
                <button
                    type="submit"
                    className = "btn btn-primary btn-block mb-3"
                >
                    Registrarse
                </button>
              
                <Link 
                    to="/auth/login"
                    className="link"
                >
                    Logueate aquí
                </Link>
            </form>
        </>
    )
}
