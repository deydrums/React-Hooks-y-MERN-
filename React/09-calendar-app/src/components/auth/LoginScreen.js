import React from 'react';
import { useDispatch } from 'react-redux';
import { startLogin } from '../../actions/auth';
import { useForm } from '../../hooks/useForm';
import './login.css';


export const LoginScreen = () => {

    const dispatch = useDispatch();

    const [formLoginValues, handleLoginInputChange, reset] = useForm({
        lemail: 'dagarcia100@gmail.com',
        lpassword: '123456'
    });

    const{ lemail, lpassword } = formLoginValues;

    const handleLogin = (e) => {
        e.preventDefault();
        dispatch(startLogin(lemail, lpassword));
    }

    return (
        
        <div className="principal-login-container">
            <div className="login-container">
                <div className="row">
                    <div className="col-md-6 login-form-1">
                        <h3>Ingreso</h3>
                        <form onSubmit={handleLogin}>
                            <div className="form-group">
                                <input 
                                    type="text"
                                    className="form-control"
                                    placeholder="Correo"
                                    name="lemail"
                                    value={lemail}
                                    onChange={handleLoginInputChange}
                                />
                            </div>
                            <div className="form-group">
                                <input
                                    type="password"
                                    className="form-control"
                                    placeholder="Contraseña"
                                    name="lpassword"
                                    value={lpassword}
                                    onChange={handleLoginInputChange}
                                />
                            </div>
                            <div className="form-group">
                                <input 
                                    type="submit"
                                    className="btnSubmit"
                                    value="Login" 
                                />
                            </div>
                        </form>
                    </div>

                    <div className="col-md-6 login-form-2">
                        <h3>Registro</h3>
                        <form>
                            <div className="form-group">
                                <input
                                    type="text"
                                    className="form-control"
                                    placeholder="Nombre"
                                />
                            </div>
                            <div className="form-group">
                                <input
                                    type="email"
                                    className="form-control"
                                    placeholder="Correo"
                                />
                            </div>
                            <div className="form-group">
                                <input
                                    type="password"
                                    className="form-control"
                                    placeholder="Contraseña" 
                                />
                            </div>

                            <div className="form-group">
                                <input
                                    type="password"
                                    className="form-control"
                                    placeholder="Repita la contraseña" 
                                />
                            </div>

                            <div className="form-group">
                                <input 
                                    type="submit" 
                                    className="btnSubmit" 
                                    value="Crear cuenta" />
                            </div>
                        </form>
                    </div>
                </div>
            </div>
        </div>
    )
}