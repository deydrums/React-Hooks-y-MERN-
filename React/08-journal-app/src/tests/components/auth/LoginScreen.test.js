import { LoginScreen } from "../../../components/auth/LoginScreen";
import {mount} from 'enzyme';
import { Provider } from "react-redux";

import configureStore from 'redux-mock-store'; //ES6 modules
import thunk from 'redux-thunk';
import { MemoryRouter } from "react-router-dom";
const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
    auth:{
        uid: {},
    },
    ui: {
        loading: false,
        msgError: null
    }
};

let store = mockStore(initState);


describe('Pruebas en <LoginScreen/>', () => {

    beforeEach(() => {
        store = mockStore(initState)
    });

    test('Debe de mostrar el componente', () => {
        const wrapper = mount(
            <MemoryRouter>
                <Provider store = {store}>
                    <LoginScreen/>
                </Provider>
            </MemoryRouter>
        );
        expect(wrapper).toMatchSnapshot();
    });
    
});
