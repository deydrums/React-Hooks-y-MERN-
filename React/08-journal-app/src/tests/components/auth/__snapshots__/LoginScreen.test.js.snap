// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen/> Debe de mostrar el componente 1`] = `
Array [
  <h3
    className="auth__title"
  >
    Iniciar sesión
  </h3>,
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="Email..."
      type="text"
      value=""
    />
    <input
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="Password..."
      type="password"
      value=""
    />
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      type="submit"
    >
      Entrar
    </button>
    <div
      className="auth__social-networks"
    >
      <p>
        Iniciar con redes sociales
      </p>
      <div
        className="google-btn"
        onClick={[Function]}
      >
        <div
          className="google-icon-wrapper"
        >
          <img
            alt="google button"
            className="google-icon"
            src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
          />
        </div>
        <p
          className="btn-text"
        >
          <b>
            Entrar con Google
          </b>
        </p>
      </div>
    </div>
    <a
      className="link"
      href="/auth/register"
      onClick={[Function]}
    >
      Registrarse aquí
    </a>
  </form>,
]
`;
